node/src/config.rs

```diff
@@ -530,6 +530,31 @@ impl Chain {
         for provider in self.providers.iter_mut() {
             provider.validate()?
         }
+
+        if !matches!(self.protocol, BlockchainKind::Substreams) {
+            let has_only_substreams_providers = self
+                .providers
+                .iter()
+                .all(|provider| matches!(provider.details, ProviderDetails::Substreams(_)));
+            if has_only_substreams_providers {
+                bail!(
+                    "{} protocol requires an rpc or firehose endpoint defined",
+                    self.protocol
+                );
+            }
+        }
+
+        // When using substreams protocol, only substreams endpoints are allowed
+        if matches!(self.protocol, BlockchainKind::Substreams) {
```

Comment:
This is equivalent to an `else`, correct?

Reply:
Yes, this was mostly to make it clearer, if you think it isn't I can refactor to improve

Reply:
No need, it's fine!